src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -438,8 +438,11 @@ internal List<UtxoStakeDescription> GetUtxoStakeDescriptions(WalletSecret wallet
         {
             var utxoStakeDescriptions = new List<UtxoStakeDescription>();
 
+            var options = (PosConsensusOptions)this.network.Consensus.Options;
+            int minDepth = options.GetStakeMinConfirmations(this.chainIndexer.Height, this.network);
+
             List<UnspentOutputReference> stakableUtxos = this.walletManager
-                .GetSpendableTransactionsInWalletForStaking(walletSecret.WalletName, 1)
+                .GetSpendableTransactionsInWalletForStaking(walletSecret.WalletName, minDepth)
```

Comment:
@zeptin can you confirm where this is filtered out later in the code? Is it in `GetBlocksCountToMaturityAsync`?

Also, if we are filtering the later, then itsn't the code that occurs after this redundant?

Reply:
@fassadlr it was in `GetUtxoStakeDescriptionsSuitableForStakingAsync`. I have made another commit that attempts to remove the introduced redundancy.

The rationale for the original change is that we don't want to hit the coinview for UTXOs we know we won't need. This does however make the computation of the immature balance later in the code slightly harder.

Reply:
I see that there are still some tests that rely on the old behaviour, which initially selects UTXOs that are not valid for staking.

Reply:
>we don't want to hit the coinview for UTXOs we know we won't need 

Is this the only reason?
Its not such a big deal as you only stake when you are at the tip and the UTXO is not hit that much anyway at that point, not to mention coin db will cache the UTXOS it fetches so subsequent calls are cached anyway.